crates/oci-distribution/src/client.rs

```diff
@@ -751,12 +753,88 @@ impl ClientProtocol {
 }
 
 /// A token granted during the OAuth2-like workflow for OCI registries.
-#[derive(serde::Deserialize, Default)]
+#[derive(Default)]
 struct RegistryToken {
-    #[serde(alias = "access_token")]
     token: String,
 }
 
+const REGISTRY_TOKEN_JSON_PRIMARY_FIELD: &str = "token";
+const REGISTRY_TOKEN_JSON_SECONDARY_FIELD: &str = "access_token";
+
+impl<'de> de::Deserialize<'de> for RegistryToken {
+    /// Custom deserialize implementation for RegistryToken.
+    ///
+    /// When requesting a scoped authorization token from a registry,
+    /// an opaque token (typically JWT) is returned in a JSON response.
+    /// Some registries place this token in the 'token' field, whereas
+    /// other registries place this token in the 'access_token' field.
+    /// In rare cases, both 'token' and 'access_token' are provided
+    /// with identical values for compatibility purposes. In this
+    /// scenario, the 'access_token' field can be safely ignored.
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct RegistryTokenVisitor;
+
+        impl<'de> de::Visitor<'de> for RegistryTokenVisitor {
+            type Value = RegistryToken;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("RegistryToken")
+            }
+
+            fn visit_map<V>(self, mut map: V) -> Result<Self::Value, V::Error>
+            where
+                V: de::MapAccess<'de>,
+            {
+                let mut token = String::new();
+                loop {
+                    let next_key_result = map.next_key::<String>();
+                    if next_key_result.is_err() {
+                        return Err(de::Error::custom("Unable to parse JSON"));
+                    }
+                    let next_key_option = next_key_result.unwrap();
+                    if next_key_option.is_none() {
+                        break; // done reading the map
+                    }
+                    let next_value_result = map.next_value::<String>();
+                    if next_value_result.is_err() {
+                        continue; // does not parse as string-to-string (e.g. 'expires_in' field)
+                    }
```

Comment:
Instead of reading the key and value separately, is it worth calling `next_entry` to get them as a tuple?  This might be more concise, but I don't know how it plays with your error/exit/continue strategy.